### PR TITLE
Add new field 'name' for sites:

### PR DESCRIPTION
**Pull request by jdrossl**
- Update create site API
- Add new API to update sites
- Add new permission to update sites
- Update API docs

https://github.com/craftercms/craftercms/issues/4192

### CODE REVIEW

File: src/main/java/org/craftercms/studio/api/v1/service/site/SiteService.java [hunk 1]
```diff
@@ -162,7 +161,7 @@ void createSiteWithRemoteOption(String siteId, String sandboxBranch, String desc
                                     String remoteName, String remoteUrl, String remoteBranch, boolean singleBranch,
                                     String authenticationType, String remoteUsername, String remotePassword,
                                     String remoteToken, String remotePrivateKey, String createOption,
-                                    Map<String, String> params, boolean createAsOrphan)
+                                    Map<String, String> params, boolean createAsOrphan, String siteName)
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/service/site/SiteService.java]: Move `siteName` to be in the correct/expected order in the parameter list.

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 2]
```diff
@@ -614,7 +610,8 @@ public void createSiteWithRemoteOption(@ValidateStringParam(name = "siteId") Str
                                            String remoteUsername, String remotePassword, String remoteToken,
                                            String remotePrivateKey,
                                            @ValidateStringParam(name = "createOption") String createOption,
-                                           Map<String, String> params, boolean createAsOrphan)
+                                           Map<String, String> params, boolean createAsOrphan,
+                                           String siteName)
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Move siteName to the correct/expected order in the parameter list.

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 3]
```diff
@@ -634,14 +631,14 @@ public void createSiteWithRemoteOption(@ValidateStringParam(name = "siteId") Str
                 logger.info("Clone from remote repository create option selected");
                 createSiteCloneRemote(siteId, sandboxBranch, description, remoteName, remoteUrl, remoteBranch,
                         singleBranch, authenticationType, remoteUsername, remotePassword, remoteToken, remotePrivateKey,
-                        params, createAsOrphan);
+                        params, createAsOrphan, siteName);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Move siteName to the correct/expected order in the parameter list.

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 4]
```diff
@@ -634,14 +631,14 @@ public void createSiteWithRemoteOption(@ValidateStringParam(name = "siteId") Str
...
                 break;
 
             case REMOTE_REPOSITORY_CREATE_OPTION_PUSH:
                 logger.info("Push to remote repository create option selected");
                 createSitePushToRemote(siteId, sandboxBranch, description, blueprintName, remoteName, remoteUrl,
                         remoteBranch, authenticationType, remoteUsername, remotePassword, remoteToken,
-                        remotePrivateKey, params, createAsOrphan);
+                        remotePrivateKey, params, createAsOrphan, siteName);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Move siteName to the correct/expected order in the parameter list.

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 5]
```diff
@@ -657,7 +654,7 @@ private void createSiteCloneRemote(String siteId, String sandboxBranch, String d
                                        String remoteUrl, String remoteBranch, boolean singleBranch,
                                        String authenticationType, String remoteUsername, String remotePassword,
                                        String remoteToken, String remotePrivateKey, Map<String, String> params,
-                                       boolean createAsOrphan)
+                                       boolean createAsOrphan, String siteName)
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Move siteName to the correct/expected order in the parameter list.

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 6]
```diff
@@ -800,7 +797,7 @@ private void createSitePushToRemote(String siteId, String sandboxBranch, String
                                         String remoteName, String remoteUrl, String remoteBranch,
                                         String authenticationType, String remoteUsername, String remotePassword,
                                         String remoteToken, String remotePrivateKey, Map<String, String> params,
-                                        boolean createAsOrphan)
+                                        boolean createAsOrphan, String siteName)
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java]: Move siteName to the correct/expected order in the parameter list.

File: src/main/java/org/craftercms/studio/impl/v2/service/marketplace/internal/MarketplaceServiceInternalImpl.java [hunk 7]
```diff
@@ -238,6 +238,6 @@ public void createSite(CreateSiteRequest request) throws RemoteRepositoryNotFoun
             result.get(PLUGIN_URL).toString(), result.get(PLUGIN_REF).toString(), false,
             RemoteRepository.AuthenticationType.NONE, null, null, null,
             null, StudioConstants.REMOTE_REPOSITORY_CREATE_OPTION_CLONE, request.getSiteParams(),
-            true);
+            true, request.getName());
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/marketplace/internal/MarketplaceServiceInternalImpl.java]: Move siteName to the correct/expected order in the parameter list.

File: src/main/resources/crafter/studio/upgrade/3.2.x/config/permission-mappings-config/permission-mappings-config-v14.xslt [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        <xsl:copy>
+            <xsl:apply-templates select="node() | @*"/>
+        </xsl:copy>
+        <!-- insert line breaks after comments -->
+        <xsl:if test="self::comment()">
+            <xsl:text>&#10;</xsl:text>
+        </xsl:if>
+    </xsl:template>
+
+    <xsl:template match="permissions/role/rule/allowed-permissions">
```
Comment (sumerjabri) [inline src/main/resources/crafter/studio/upgrade/3.2.x/config/permission-mappings-config/permission-mappings-config-v14.xslt:40]: What about `site-admins`? They should also be able to edit the site name and site description.

File: src/main/webapp/default-site/scripts/api/SiteServices.groovy [hunk 9]
```diff
@@ -69,11 +69,11 @@ class SiteServices {
 
     static createSiteWithRemoteOption(context, siteId, sandboxBranch, description, blueprint, remoteName, remoteUrl,
                                       remoteBranch, singleBranch, authenticationType, remoteUsername, remotePassword,
-                                      remoteToken, remotePrivateKey, createOption, params, createAsOrphan) {
+                                      remoteToken, remotePrivateKey, createOption, params, createAsOrphan, name) {
         def siteServicesImpl = ServiceFactory.getSiteServices(context)
         return siteServicesImpl.createSiteWithRemoteOption(siteId, sandboxBranch, description, blueprint, remoteName,
                 remoteUrl, remoteBranch, singleBranch,authenticationType, remoteUsername, remotePassword, remoteToken,
-                remotePrivateKey, createOption, params, createAsOrphan)
+                remotePrivateKey, createOption, params, createAsOrphan, name)
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/api/SiteServices.groovy]: Move siteName to the correct/expected order in the parameter list. (change `name` to siteName)

File: src/main/webapp/default-site/scripts/api/SiteServices.groovy [hunk 10]
```diff
@@ -69,11 +69,11 @@ class SiteServices {
 
     static createSiteWithRemoteOption(context, siteId, sandboxBranch, description, blueprint, remoteName, remoteUrl,
                                       remoteBranch, singleBranch, authenticationType, remoteUsername, remotePassword,
-                                      remoteToken, remotePrivateKey, createOption, params, createAsOrphan) {
+                                      remoteToken, remotePrivateKey, createOption, params, createAsOrphan, name) {
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/api/SiteServices.groovy]: Move siteName to the correct/expected order in the parameter list. (change `name` to siteName)

File: src/main/webapp/default-site/scripts/api/impl/site/SpringSiteServices.groovy [hunk 11]
```diff
@@ -68,11 +68,11 @@ class SpringSiteServices {
 
     def createSiteWithRemoteOption(siteId, sandboxBranch, description, blueprint, remoteName, remoteUrl, remoteBranch,
                                    singleBranch, authenticationType, remoteUsername, remotePassword, remoteToken,
-                                   remotePrivateKey, createOption, params, createAsOrphan) {
+                                   remotePrivateKey, createOption, params, createAsOrphan, siteName) {
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/api/impl/site/SpringSiteServices.groovy]: Move siteName to the correct/expected order in the parameter list.

File: src/main/webapp/default-site/scripts/api/impl/site/SpringSiteServices.groovy [hunk 12]
```diff
@@ -68,11 +68,11 @@ class SpringSiteServices {
...
         def springBackedService = this.context.applicationContext.get(SITE_SERVICES_BEAN)
         return springBackedService.createSiteWithRemoteOption(siteId, sandboxBranch, description, blueprint, remoteName,
                 remoteUrl, remoteBranch, singleBranch, authenticationType, remoteUsername, remotePassword, remoteToken,
-                remotePrivateKey, createOption, params, createAsOrphan)
+                remotePrivateKey, createOption, params, createAsOrphan, siteName)
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/api/impl/site/SpringSiteServices.groovy]: Move siteName to the correct/expected order in the parameter list.

File: src/main/webapp/default-site/scripts/rest/api/1/site/create.post.groovy [hunk 13]
```diff
@@ -209,7 +215,7 @@ try {
             } else {
                 SiteServices.createSiteWithRemoteOption(context, siteId, sandboxBranch, description, blueprint,
                         remoteName, remoteUrl, remoteBranch, singleBranch, authenticationType, remoteUsername,
-                        remotePassword, remoteToken, remotePrivateKey, createOption, siteParams, createAsOrphan)
+                        remotePassword, remoteToken, remotePrivateKey, createOption, siteParams, createAsOrphan, name)
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/rest/api/1/site/create.post.groovy]: Move siteName to the correct/expected order in the parameter list.

File: src/main/webapp/default-site/scripts/rest/api/1/site/create.post.groovy [hunk 14]
```diff
@@ -199,7 +205,7 @@ try {
         def context = SiteServices.createContext(applicationContext, request)
         try {
             if (!useRemote) {
-                SiteServices.createSiteFromBlueprint(context, blueprint, siteId, siteId, sandboxBranch, description,
+                SiteServices.createSiteFromBlueprint(context, blueprint, name, siteId, sandboxBranch, description,
```
Comment (sumerjabri) [inline src/main/webapp/default-site/scripts/rest/api/1/site/create.post.groovy]: Should be siteId then siteName, throughout the system in terms of parameter order.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please also update the blueprints to have the new permission assigned to site-admins

File: src/main/webapp/default-site/scripts/api/SiteServices.groovy (hunk 10 above)
Comment (jdrossl) [reply src/main/webapp/default-site/scripts/api/SiteServices.groovy]: should the parameter in the request also be `siteName` instead of `name`?

Comment (sumerjabri) [reply src/main/webapp/default-site/scripts/api/SiteServices.groovy]: No need to change the external API in V1. I'm just looking to change the internal names to be consistent.

File: src/main/resources/crafter/studio/upgrade/3.2.x/config/permission-mappings-config/permission-mappings-config-v14.xslt (hunk 8 above)
Comment (sumerjabri) [reply src/main/resources/crafter/studio/upgrade/3.2.x/config/permission-mappings-config/permission-mappings-config-v14.xslt:40]: @jdrossl has this been addressed?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/resources/crafter/studio/upgrade/3.2.x/config/permission-mappings-config/permission-mappings-config-v14.xslt (hunk 8 above)
Comment (jdrossl) [reply src/main/resources/crafter/studio/upgrade/3.2.x/config/permission-mappings-config/permission-mappings-config-v14.xslt:40]: yeah https://github.com/craftercms/studio/pull/2202/files?file-filters%5B%5D=.xslt#diff-5e8e31cb61906b9a672c96fa634071c8R53

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**